src/i18n/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+find_program(GETTEXT_MSGFMT_EXECUTABLE msgfmt)
+if(NOT GETTEXT_MSGFMT_EXECUTABLE)
+    message(FATAL_ERROR "msgfmt not found")
+endif()
+
+find_program(GETTEXT_MSGMERGE_EXECUTABLE msgmerge)
+if(NOT GETTEXT_MSGMERGE_EXECUTABLE)
+    message(FATAL_ERROR "msgmerge not found")
+endif()
+
+find_program(GETTEXT_MSGATTRIB_EXECUTABLE msgattrib)
+if(NOT GETTEXT_MSGATTRIB_EXECUTABLE)
+    message(FATAL_ERROR "msgattrib not found")
+endif()
+
+SET(TRANSLATIONS
+    de
+    en
+    es
+    it
+    ru
+)
+
+foreach(_trans ${TRANSLATIONS})
+add_custom_command(
+   COMMENT "Generate Locale '${_trans}'"
+   OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/Locale/${_trans}.po"
+   POST_BUILD
+   COMMAND mkdir -p Locale/${_trans}/LC_MESSAGES
+   COMMAND ${GETTEXT_MSGATTRIB_EXECUTABLE} --translated "${CMAKE_CURRENT_SOURCE_DIR}/Locale/${_trans}.po" -o "${CMAKE_CURRENT_SOURCE_DIR}/Locale/${_trans}.po"
+   COMMAND ${GETTEXT_MSGMERGE_EXECUTABLE} --update --no-fuzzy-matching "${CMAKE_CURRENT_SOURCE_DIR}/Locale/${_trans}.po" "${CMAKE_CURRENT_SOURCE_DIR}/Locale/en.po"
```

Comment:
" `Line 30` removes *empty translations* (untranslated)"
Looks good to me

"`Line 31` merges *missing translations* from the `en.po`"
Line 31 uses msgmerge
Here is the [description of msgmerge](https://www.gnu.org/software/gettext/manual/html_node/msgmerge-Invocation.html#msgmerge-Invocation). It states that any comments and translations form the second file are discarded.

Should msgcat be used in stead?
E.g. `msgcat --use-first fr.po en.po -o fr.po`
Of course this would have to be changed slightly for cmake

"`Line 32` merges every *new/missing/empty translations* into it from the `template.po`"
Okay, that seems right.

It adds empty translations to %{_trans}.po for each non existing translation. This is unnecessary if it is being generated to be used as input to a .mo file because empty translations are left out of the .mo file.

But this does makes sense if the .po file is being generated for translators as well, which I assume it is.

TL;DR
Should msgcat be used instead of msgmerge on L31?
Is L32 generating .po files for translators or are they made elsewhere?